### PR TITLE
README.md cleanup for 1.3.2

### PR DESCRIPTION
**Pull request by chrisvanheuveln**
* Moved Document Workflow Map to a Documentation Guide table at the end

* Combined Overview and Module Description

* Updated Usage section

* Moved info from the Limitations section into the Platform Support Matrix, which moved after the Resource Reference tables

* Created Example Manifests Section

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,96 +1,119 @@
...
-1. [Learning Resources](#learning-resources)
-
-
+   * [Resource Platform Support Matrix](#resource-platform-support-matrix)
+1. [Documentation Guide](#documentation-guide)
 
-## Overview
+## <a href='module-description'>Module Description</a>
 
 The ciscopuppet module allows a network administrator to manage Cisco Network Elements using Puppet. This module bundles a set of Puppet Types, providers, Beaker Tests, Sample Manifests and Installation Tools for effective network management.  The  resources and capabilities provided by this Puppet Module will grow with contributions from Cisco, Puppet Labs and the open source community.
```
Comment (mikewiebe) [inline README.md]: Can you make `providers` uppercase in the second sentence?  `This module bundles a set of Puppet Types, Providers, Beaker...`

File: README.md [hunk 2]
```diff
@@ -1,96 +1,119 @@
...
 
-The Cisco Network Elements and Operating Systems managed by this Puppet Module are continuously expanding. Please refer to the [Limitations](#limitations) section for details on currently supported hardware and software.
-The Limitations section also provides details on compatible Puppet Agent and Puppet Master versions.
+The Cisco Network Elements and Operating Systems managed by this Puppet Module are continuously expanding. See [Cisco Resource Platform Support](#cisco-resource-platform-support) for a list of currently supported hardware and software.
```
Comment (mikewiebe) [inline README.md]: `Cisco Resource Platform Support` link is broken.

File: README.md [hunk 3]
```diff
@@ -1,96 +1,119 @@
...
 
 This GitHub repository contains the latest version of the ciscopuppet module source code. Supported versions of the ciscopuppet module are available at Puppet Forge. Please refer to [SUPPORT.md](SUPPORT.md) for additional details.
```
Comment (mikewiebe) [inline README.md]: We should make sure that `[SUPPORT.md]` links to master so that it works properly on puppet forge.

File: README.md [hunk 4]
```diff
@@ -1,96 +1,119 @@
...
 
-Contributions to this Puppet Module are welcome. Guidelines on contributions to the module are captured in [CONTRIBUTING.md](CONTRIBUTING.md)
+##### Dependencies
+
+The `ciscopuppet` module has a dependency on the [`cisco_node_utils`](https://rubygems.org/gems/cisco_node_utils) ruby gem. See the **Setup** section that follows for more information on `cisco_node_utils`.
+
+##### Contributing
+
+Contributions to the `ciscopuppet` module are welcome. See [CONTRIBUTING.md](CONTRIBUTING.md) for guidelines.
```
Comment (mikewiebe) [inline README.md]: Here `[CONTRIBUTING.md]` and elsewhere we should make sure that links to other docs in the repo link to master.

File: README.md [hunk 5]
```diff
@@ -1,96 +1,119 @@
...
+```bash
+puppet module install puppetlabs-ciscopuppet
+```
+
+For more information on Puppet module installation see [Puppet Labs: Installing Modules](https://docs.puppetlabs.com/puppet/latest/reference/modules_installing.html)
 
-### Puppet Master
+##### The `puppetlabs-netdev_stdlib` module
 
-The `ciscopuppet` module must be installed on the Puppet Master server. Please see [Puppet Labs: Installing Modules](https://docs.puppetlabs.com/puppet/latest/reference/modules_installing.html) for general information on Puppet module installation.
```
Comment (mikewiebe) [inline README.md]: I would re-word this as follows:
```
PuppetLabs provides NetDev resource support for Cisco Nexus devices with their...
```

File: README.md [hunk 6]
```diff
@@ -1,96 +1,119 @@
...
+~~~bash
+cd /etc/puppetlabs/code/environments/production/modules/ciscopuppet/
+cp examples/install.pp  manifests/
+~~~
+
+* Next, update `site.pp` to use the install class
+
+**Example**
+
+~~~
```
Comment (mikewiebe) [inline README.md]: Use `  ~~~puppet `

File: README.md [hunk 7]
```diff
@@ -1,96 +1,119 @@
...
+The preceding configuration will cause the next `puppet agent` run to automatically download the current `cisco_node_utils` gem from <https://rubygems.org/gems/cisco_node_utils> and install it on the node.
+
+##### Optional Parameters for `ciscopuppet::install`
+
+  * Override the default rubygems repository to use a custom repository
+  * Provide a proxy server
+
+**Example**
+
+~~~
```
Comment (mikewiebe) [inline README.md]: Use `  ~~~puppet `

File: README.md [hunk 8]
```diff
@@ -1,96 +1,119 @@
...
+node default
+  include ciscopuppet::install
```
Comment (mikewiebe) [inline README.md]: The `include ciscopuppet::install` is not needed in this case.  You can remove line 84.

File: README.md [hunk 9]
```diff
@@ -1,96 +1,119 @@
...
+    repo  => 'http://gemserver.domain.com:8808',
+    proxy => 'http://proxy.domain.com:8080',
+  }
+end
+~~~
+
+##### Gem Persistence
 
-This module has dependencies on the [`cisco_node_utils`](https://rubygems.org/gems/cisco_node_utils) ruby gem. After installing the Puppet Agent software, use Puppet's built-in [`Package`](https://github.com/cisco/cisco-network-puppet-module/blob/master/examples/install.pp#L17) provider to install the gem.
+Once installed, the GEM will remain persistent across system reloads within the Guestshell or OAC environments; however, the bash-shell environment does not share this persistent behavior, in which case the `ciscopuppet::install` helper class automatically downloads and re-installs the gem after each system reload.
```
Comment (mikewiebe) [inline README.md]: Perhaps provide a pointer to our Guestshell and OAC docs?  This is the first time the terms are introduced, so some context might be nice.

File: README.md [hunk 10]
```diff
@@ -1,96 +1,119 @@
...
+cp examples/install.pp  manifests/
+~~~
+
+* Next, update `site.pp` to use the install class
+
+**Example**
+
+~~~
+node default
+  include ciscopuppet::install
```
Comment (mikewiebe) [inline README.md]: This should use the following syntax right?
~~~puppet
node 'default' {
  include ciscopuppet::install
}
~~~
https://github.com/cisco/cisco-network-puppet-module/blob/develop/examples/demo_site_cisco.pp#L17

File: README.md [hunk 11]
```diff
@@ -1,96 +1,119 @@
...
+
+  * Override the default rubygems repository to use a custom repository
+  * Provide a proxy server
+
+**Example**
+
+~~~
+node default
+  include ciscopuppet::install
+  class {'ciscopuppet::install':
```
Comment (mikewiebe) [inline README.md]: This should use the following syntax right?
~~~puppet
node 'default' {
  class {'ciscopuppet::install':
    repo  => 'http://gemserver.domain.com:8808',
    proxy => 'http://proxy.domain.com:8080',
}
~~~
https://github.com/cisco/cisco-network-puppet-module/blob/develop/examples/demo_site_cisco.pp#L17

File: README.md [hunk 12]
```diff
@@ -1,96 +1,119 @@
...
 
-A helper class [`ciscopuppet::install`](https://github.com/cisco/cisco-network-puppet-module/blob/master/examples/demo_all_cisco.pp#L19) is provided in the examples subdirectory of this module.  Simply add an `include ciscopuppet::install` statement at the beginning of the manifest to install the latest `cisco_node_utils` gem from rubygems.org. Including the aforementioned class with [`additional parameters`](https://github.com/cisco/cisco-network-puppet-module/blob/master/examples/demo_all_cisco.pp#L24) overrides the default rubygems.org repository with a custom repository.
+## <a href='example-manifests'>Example Manifests</a>
 
-For Puppet Agents running within the GuestShell or OAC environment, the installed GEM remains persistent across system reloads, however, agents running in the NX-OS bash-shell environment will automatically download and reinstall the GEM after a system reload.
+Example manifests for all `ciscopuppet` providers can be found in the demo files in the [examples/cisco](https://github.com/cisco/cisco-network-puppet-module/tree/develop/examples/cisco) directory.
```
Comment (mikewiebe) [inline README.md]: Use `master` instead of `develop`

File: README.md [hunk 13]
```diff
@@ -425,6 +346,112 @@ The following resources include cisco types and providers along with cisco provi
...
+| [cisco_stp_global](#type-cisco_stp_global)                 | ✅* | ✅* | ✅* | ✅* | ✅ | ✅ | \*[caveats](#cisco_stp_global-caveats) |
+| [cisco_snmp_community](#type-cisco_snmp_community)         | ✅  | ✅  | ✅  | ✅  | ✅ | ✅ |
+| [cisco_snmp_group](#type-cisco_snmp_group)                 | ✅  | ✅  | ✅  | ✅  | ✅ | ✅ |
+| [cisco_snmp_server](#type-cisco_snmp_server)               | ✅  | ✅  | ✅  | ✅  | ✅ | ✅ |
+| [cisco_snmp_user](#type-cisco_snmp_user)                   | ✅  | ✅  | ✅  | ✅  | ✅ | ✅ |
+| [cisco_tacacs_server](#type-cisco_tacacs_server)           | ✅  | ✅  | ✅  | ✅  | ✅ | ✅ |
+| [cisco_tacacs_server_host](#type-cisco_tacacs_server_host) | ✅  | ✅  | ✅  | ✅  | ✅ | ✅ |
+| [cisco_vdc](#type-cisco_vdc)                               | :heavy_minus_sign: | :heavy_minus_sign: | :heavy_minus_sign: | :heavy_minus_sign: | ✅ | :heavy_minus_sign: |
+| [cisco_vlan](#type-cisco_vlan)                             | ✅* | ✅* | ✅  | ✅  | ✅ | ✅ | \*[caveats](#cisco_vlan-caveats) |
+| [cisco_vpc_domain](#type-cisco_vpc_domain)                 | ✅* | ✅* | ✅* | ✅* | ✅* | ❌	| \*[caveats](#cisco_vpc_domain-caveats) |
```
Comment (mikewiebe) [inline README.md]: We still have a ❌ for vcp_domain so we need to update the key above.

File: README.md (hunk 1 above)
Comment (chrisvanheuveln) [inline README.md]: done

File: README.md (hunk 2 above)
Comment (chrisvanheuveln) [inline README.md]: fixed

File: README.md (hunk 3 above)
Comment (chrisvanheuveln) [inline README.md]: Fixed. Updated all other links to use master as well.

File: README.md (hunk 4 above)
Comment (chrisvanheuveln) [inline README.md]: yep, updated.

File: README.md (hunk 5 above)
Comment (chrisvanheuveln) [inline README.md]: re-worded.

File: README.md (hunk 6 above)
Comment (chrisvanheuveln) [inline README.md]: fixed

File: README.md (hunk 7 above)
Comment (chrisvanheuveln) [inline README.md]: done

File: README.md (hunk 8 above)
Comment (chrisvanheuveln) [inline README.md]: yep, fixed

File: README.md (hunk 11 above)
Comment (chrisvanheuveln) [inline README.md]: fixed

File: README.md (hunk 9 above)
Comment (chrisvanheuveln) [inline README.md]: done

File: README.md (hunk 10 above)
Comment (chrisvanheuveln) [inline README.md]: fixed

File: README.md (hunk 12 above)
Comment (chrisvanheuveln) [inline README.md]: done throughout

File: README.md (hunk 13 above)
Comment (chrisvanheuveln) [inline README.md]: The vpc.yaml file excludes N8k. Are you sure they have VPC support on that box?

File: README.md [hunk 14]
```diff
@@ -1,96 +1,120 @@
...
 
-The `ciscopuppet` module must be installed on the Puppet Master server. Please see [Puppet Labs: Installing Modules](https://docs.puppetlabs.com/puppet/latest/reference/modules_installing.html) for general information on Puppet module installation.
+##### The `puppetlabs-netdev_stdlib` module
+
+PuppetLabs provides NetDev resource support for Cisco Nexus devices with their [`puppetlabs-netdev-stdlib`](https://forge.puppet.com/puppetlabs/netdev_stdlib) module. Installing the `ciscopuppet` module automatically installs both the `ciscopuppet` and `netdev_stdlib` modules.
+
+#### Puppet Agent
 
-### Puppet Agent
 The Puppet Agent requires installation and setup on each device. Agent setup can be performed as a manual process or it may be automated. For more information please see the [README-agent-install.md](docs/README-agent-install.md) document for detailed instructions on agent installation and configuration on Cisco Nexus devices.
```
Comment (mikewiebe) [inline README.md]: Update `[README-agent-install.md]` to point to master.

Comment (chrisvanheuveln) [inline README.md]: Fixed

**Pull request merged**